views/client.go

```diff
@@ -90,3 +191,141 @@ func startViewsQuery(playbackID, playbackRecordingID string) string {
 		queryID,
 	)
 }
+
+var viewershipBreakdownFields = map[string]string{
+	"device":      "device",
+	"os":          "os",
+	"user_agent":  "user_agent_family",
+	"continent":   "playback_continent_name",
+	"country":     "playback_country_name",
+	"subdivision": "playback_subdivisions_name",
+	"timezone":    "playback_timezone",
+	"viewer_id":   "viewer_id",
+}
+
+var allowedTimeSteps = map[string]bool{
+	"hour":  true,
+	"day":   true,
+	"week":  true,
+	"month": true,
+	"year":  true,
+}
+
+type LDSMStateRow struct {
+	TimeInterval time.Time `bigquery:"time_interval"`
+	PlaybackID   string    `bigquery:"playback_id"`
+
+	// breakdown fields
+	Device      string `bigquery:"device"`
+	OS          string `bigquery:"os"`
+	UserAgent   string `bigquery:"user_agent_family"`
```

Comment:
this one doesn't exist in this iteration. the ua-parser through js was a bit different from the python one i was using

Reply:
there are the 4 fields that should be added 
-- device_type 
-- browser
-- browser_engine
-- cpu